src/tools/business/gcal_tool.py

```
@@ -477,13 +678,82 @@ def _intersect(a: list[tuple[datetime, datetime]], b: list[tuple[datetime, datet
+                            tn_min = self._normalize_datetime_to_calendar_tz(time_min, tz_name_k)
+                            tn_max = self._normalize_datetime_to_calendar_tz(time_max, tz_name_k)
+                            free_blocks_dt = _working_hours_mask(
+                                tn_min, tn_max, tz_name_k,
+                                work_start_hour, work_end_hour, work_days_set,
+                            )
+                            busy_blocks_dt: list[tuple[datetime, datetime]] = []
+                            for s_iso, e_iso in busy_isos:
+                                try:
+                                    busy_blocks_dt.append((self._parse_iso(s_iso), self._parse_iso(e_iso)))
+                                except Exception:
+                                    continue
```

Ruff (0.15.11)

[error] `try`-`except`-`continue` detected, consider logging the exception

(S112)